File: e-commerce/ShopingApi/ShopingApi/ViewModels/Product/ProductDetails.cs
using ShopingApi.Models;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace ShopingApi.ViewModels.Product
{
    public class ProductDetails
    {
        
        public int Id { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }

        public List<Size>? Size { get; set; }

        public List<Color>? Colors { get; set; }
        public int Quantity { get; set; } 
        public decimal Price { get; set; }

        public string? ImageURL { get; set; }
        public string productTypeId { get; set; }


        public string Category { get; set; }

        // Navigation property for ProductType
        public string? ProductType { get; set; }
    }
}

File: e-commerce/ShopApp_ng/src/app/Models/productViewModel/product-details.ts
import { Color } from "./color_size/color";
import { Size } from "./color_size/size";

export class ProductDetails {
    id: number  | undefined;
    name: string | undefined;
    description?: string  | undefined;
    sizes?: Size[]  | undefined;
    colors?: Color[]  | undefined;
    quantity: number =0 ;
    price: number  =1;
    imageURL?: string  | undefined;
    productTypeId: string  | undefined;
    category: string  | undefined;
    productType?: string  | undefined;
    
}
